### PR TITLE
implemented different image list loading by piece data and example fo…

### PR DESCRIPTION
**Pull request by aleksey-alifer**
…r it

### CODE REVIEW

File: src/BackgroundImageEditor.js [hunk 1]
```diff
@@ -42,7 +42,12 @@ export default class RedaxtorBackgroundEditor extends Component {
             bgRepeat: this.targetDiv.style.backgroundRepeat,
             bgSize: this.targetDiv.style.backgroundSize,
             bgPosition: this.targetDiv.style.backgroundPosition,
-            alt: this.targetDiv.title || ""
+            alt: this.targetDiv.title || "",
+            pieceData: {
```
Comment (AndrewKirkovski) [inline src/BackgroundImageEditor.js]: please rename it to pieceRef and add "dataset" to data passed
dataset is user-specified data and may be usefull in abstract scenario

File: src/ImageTagEditor.js [hunk 2]
```diff
@@ -49,7 +49,12 @@ export default class RedaxtorImageTag extends Component {
             url: this.targetImg.src,
             alt: this.targetImg.alt || "",
             width: this.targetImg.width,
-            height: this.targetImg.height
+            height: this.targetImg.height,
+            pieceData: {
```
Comment (AndrewKirkovski) [inline src/ImageTagEditor.js]: please rename it to pieceRef and add "dataset" to data passed
dataset is user-specified data and may be usefull in abstract scenario

File: src/imageManager/ImageManager.js [hunk 3]
```diff
@@ -207,6 +213,9 @@ export default class ImageManager extends Component {
         (!data.alt && data.url) && (data.alt = "");
         this.setState(data);
         data.url && this.getImageSize(data, !!data.width);
+        if(data.pieceData){
```
Comment (AndrewKirkovski) [inline src/imageManager/ImageManager.js]: on what condition we don't have pieceData here? We should always have it, even from rich text image inserting.
b.t.w. rich text image insert functionality is not passing it currently, but it should. Add it there too

**Review by AndrewKirkovski — CHANGES_REQUESTED** [review]

File: src/BackgroundImageEditor.js (hunk 1 above)
Comment (aleksey-alifer) [reply src/BackgroundImageEditor.js]: done

File: src/ImageTagEditor.js (hunk 2 above)
Comment (aleksey-alifer) [reply src/ImageTagEditor.js]: done

File: src/imageManager/ImageManager.js (hunk 3 above)
Comment (aleksey-alifer) [reply src/imageManager/ImageManager.js]: we've disscued it on the skype. It done

**Pull request merged**